RecoEgamma/EgammaElectronProducers/plugins/GsfElectronBaseProducer.cc

```diff
@@ -40,69 +45,36 @@ void GsfElectronBaseProducer::fillDescription( edm::ParameterSetDescription & de
   desc.add<std::string>("outputEGMPFValueMap", "");
 
   // steering
+  desc.add<bool>("gedElectronMode", true);
+  desc.add<bool>("useCombinationRegression", true);
+  desc.add<bool>("ecalDrivenEcalEnergyFromClassBasedParameterization", false);
+  desc.add<bool>("ecalDrivenEcalErrorFromClassBasedParameterization", false);
+  desc.add<bool>("applyPreselection", true);
+  desc.add<bool>("useEcalRegression", true);
+  desc.add<bool>("applyAmbResolution", false);
   desc.add<bool>("useGsfPfRecTracks", true);
   desc.add<bool>("pureTrackerDrivenEcalErrorFromSimpleParameterization", true);
   desc.add<unsigned int>("ambSortingStrategy", 1);
   desc.add<unsigned int>("ambClustersOverlapStrategy", 1);
   desc.add<bool>("addPflowElectrons", true); // this one should be transfered to the "core" level
 
   // Ecal rec hits configuration
-  desc.add<std::vector<std::string>>("recHitFlagsToBeExcludedBarrel", {
-    "kFaultyHardware",
-    "kTowerRecovered",
-    "kDead",
-  });
-  desc.add<std::vector<std::string>>("recHitFlagsToBeExcludedEndcaps", {
-    "kFaultyHardware",
-    "kNeighboursRecovered",
-    "kTowerRecovered",
-    "kDead",
-    "kWeird",
-  });
-  desc.add<std::vector<std::string>>("recHitSeverityToBeExcludedBarrel", {
-    "kWeird",
-    "kBad",
-    "kTime",
-  });
-  desc.add<std::vector<std::string>>("recHitSeverityToBeExcludedEndcaps", {
-    "kWeird",
-    "kBad",
-    "kTime",
-  });
+  desc.add<std::vector<std::string>>("recHitFlagsToBeExcludedBarrel");
+  desc.add<std::vector<std::string>>("recHitFlagsToBeExcludedEndcaps");
+  desc.add<std::vector<std::string>>("recHitSeverityToBeExcludedBarrel");
+  desc.add<std::vector<std::string>>("recHitSeverityToBeExcludedEndcaps");
 
   // Isolation algos configuration
   {
     edm::ParameterSetDescription psd0;
     {
       edm::ParameterSetDescription psd1;
-      psd1.add<int>("minHits", -1);
-      psd1.add<std::vector<std::string>>("algosToReject", {
-        "jetCoreRegionalStep",
-      });
-      psd1.add<double>("minDEta", 0.015);
-      psd1.add<double>("maxDZ", 0.2);
-      psd1.add<double>("maxDPtPt", -1);
-      psd1.add<std::vector<std::string>>("allowedQualities", {});
-      psd1.add<double>("maxDR", 0.4);
-      psd1.add<double>("minDR", 0.0);
-      psd1.add<int>("minPixelHits", -1);
-      psd1.add<double>("minPt", 0.7);
+      psd1.setUnknown();
```

Comment:
If you don't set it "unknown", the individual parameters are still checked for correctness (misspelling, etc.)

Reply:
Thank you that means I should just remove this line. I was mistakenly thinking if I don't do `setUnknown`. The validation would not expect any parameters for this set and will complain because there obviously will be some.

Reply:
Well, my suggestion was to define one of these psd1 (even if they will be further modified in the actual python configs), so that the "minDEta", "maxDZ", etc. parameter names are already known to the configuration validation.

Reply:
Ok! Now I finally get what you mean.